app/src/retrieve.py

```diff
@@ -1,24 +1,22 @@
 import logging
 from typing import Sequence
 
-from sentence_transformers import SentenceTransformer
 from sqlalchemy import select
 
-import src.adapters.db as db
 from src.db.models.document import Chunk, ChunkWithScore, Document
+from src.shared import get_app_config
 
 logger = logging.getLogger(__name__)
 
 
 def retrieve_with_scores(
-    db_session: db.Session,
-    embedding_model: SentenceTransformer,
     query: str,
     k: int = 5,
     **filters: Sequence[str] | None,
 ) -> Sequence[ChunkWithScore]:
     logger.info("Retrieving context for %r", query)
 
+    embedding_model = get_app_config().sentence_transformer
```

Comment:
nit: does get_app_config() need to be called everytime or can we store the results in a variable?

Reply:
Good point!

Reply:
I had to make more changes that anticipated, especially around login, so test the login mechanism carefully and check the deployment after this merges.